### PR TITLE
Run the integration test for the dotnet part and sign DLLs

### PR DESCRIPTION
**Pull request by csaba-sagi-sonarsource**

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -179,13 +180,12 @@ build_and_release_dotnet_task:
     - source cirrus-env QA
     - PowerShell -NonInteractive -NoProfile -File "${PIPELINE_SCRIPTS_DIR}\\build_and_pack_solution.ps1"
     - PowerShell -NonInteractive -NoProfile -File "${PIPELINE_SCRIPTS_DIR}\\extract_project_version.ps1"
-  unit_test_script:
```
Comment (csaba-sagi-sonarsource) [inline .cirrus.yml:182]: Moved the UT and IT test execution to the same file not to evaluate if signing should be done for both separately.

File: sonar-text-dotnet/pipeline_scripts/execute_tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,24 @@
+$signAssembly = "$env:CIRRUS_BRANCH" -eq "master" -or "$env:CIRRUS_BRANCH".startsWith("branch-")
+Write-Host "Should the assembly be signed: $signAssembly"
+
+Write-Host "Executing unit tests"
+if($signAssembly)
```
Comment (csaba-sagi-sonarsource) [inline sonar-text-dotnet/pipeline_scripts/execute_tests.ps1:5]: AltCoverStrongNameKey should be only set if the DLLs are signed.

File: sonar-text-dotnet/pipeline_scripts/build_and_pack_solution.ps1 [hunk 3]
```diff
@@ -22,6 +22,20 @@ jf dotnet-config --global --server-id-resolve repox --repo-resolve $nugetFeed
 $signAssembly = "$env:CIRRUS_BRANCH" -eq "master" -or "$env:CIRRUS_BRANCH".startsWith("branch-")
 Write-Host "Should the assembly be signed: $signAssembly"
 
+Write-Host "Locating SignTool.exe"
+$searchRoot = ${env:ProgramFiles(x86)} + '\Windows Kits\10\bin\10*'
+$exeName = 'signtool.exe'
+$signtool = Get-ChildItem -Path $searchRoot -Filter $exeName -Recurse -ErrorAction SilentlyContinue -Force | Select -Last 1
+if (!$signtool)
+{
+throw 'Unable to find ' + $exeName + ' under ' + $searchRoot
```
Comment (costin-zaharia-sonarsource) [inline sonar-text-dotnet/pipeline_scripts/build_and_pack_solution.ps1:31]: pedantic: indentation

File: sonar-text-dotnet/pipeline_scripts/build_and_pack_solution.ps1 [hunk 4]
```diff
@@ -22,6 +22,20 @@ jf dotnet-config --global --server-id-resolve repox --repo-resolve $nugetFeed
...
+}
+
+Write-Host 'Resolving paths '
+$env:PFX_PATH = resolve-path ${env:PFX_PATH}
```
Comment (costin-zaharia-sonarsource) [inline sonar-text-dotnet/pipeline_scripts/build_and_pack_solution.ps1:35]: I have mixed feelings about overriding the environment variables.
I would go for:
```suggestion
$pfxPath = resolve-path $env:PFX_PATH
```

File: sonar-text-dotnet/pipeline_scripts/execute_tests.ps1 (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply sonar-text-dotnet/pipeline_scripts/execute_tests.ps1:5]: Was the build failing because I forgot to set `AltCoverStrongNameKey` property?

Comment (csaba-sagi-sonarsource) [reply sonar-text-dotnet/pipeline_scripts/execute_tests.ps1:5]: Yes, the same thing that is happening currently on master was happening.

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**